python/ray/serve/_private/deployment_state.py

```diff
@@ -1368,34 +1401,27 @@ def _set_target_state(self, target_info: DeploymentInfo) -> None:
             ):
                 ServeUsageTag.NUM_REPLICAS_LIGHTWEIGHT_UPDATED.record("True")
 
-        self._target_state = target_state
-        self._curr_status_info = DeploymentStatusInfo(
-            self.deployment_name, DeploymentStatus.UPDATING
-        )
-        self._replica_constructor_retry_counter = 0
-        self._backoff_time_s = 1
-
-        app_msg = f" in application '{self.app_name}'" if self.app_name else ""
-        logger.info(
-            f"Deploying new version of deployment {self.deployment_name}{app_msg}."
-        )
-
-    def _set_target_state_autoscaling(self, num_replicas: int) -> None:
-        """Update the target number of replicas based on an autoscaling decision.
-
-        This differs from _set_target_state because we are updating the
-        target number of replicas base on an autoscaling decision and
-        not a redeployment. This only changes the target num_replicas,
-        and doesn't change the current deployment status.
-        """
-
-        new_info = copy(self._target_state.info)
-        new_info.set_autoscaled_num_replicas(num_replicas)
-        new_info.version = self._target_state.version.code_version
-
-        target_state = DeploymentTargetState.from_deployment_info(new_info)
+        # Determine whether the updated target state is simply upscaling/downscaling
+        is_scale = target_state.compare_state_excluding_num_replicas(self._target_state)
+        if self._target_state.info and is_scale:
+            new, old = (target_state.num_replicas, self._target_state.num_replicas)
+            assert new != old
+            scaling_decision = (
+                DeploymentStatus.UPSCALING
+                if new > old
+                else DeploymentStatus.DOWNSCALING
+            )
```

Comment:
Where's the logic that determines the application status based on these deployment statuses? I think I missed it.

I want to confirm that the application status is `DEPLOYING` while deployments are outside the `min_replicas`/`max_replicas` bounds and then become `RUNNING` as soon as they enter those bounds, even if they're still autoscaling.

Reply:
it's in `application_state`.

I also am not sure if this is handled right now, I don't believe so.

Reply:
Yup this is not handled right now, I will add it in a follow up PR! Want to get this one in quickly to unblock folks.